### PR TITLE
[poc] Add PoC env

### PR DESCRIPTION
**Pull request by AntonVanAssche**
| Hostname | IP           | Services  |
|----------|--------------|----------|
| srv1     | 172.16.128.1 | Bind9, Node Exporter    |
| srv2     | 172.16.128.2 | Apache, Node Exporter    |
| srv3     | 172.16.128.3 | Mardiadb, Node Exporter    |
| srv4     | 172.16.128.4 | Prometheus, Node Exporter    |
| srv5     | 172.16.128.5 | Samba   |
| control     | 172.16.128.253 | Ansible    |

## Requirements

- vagrant
- VirtualBox

## Usage

```console
$ vagrant up
```

## How to test

The website configured on `srv2` should be available on the host machine at `http://172.16.128.2`, displaying the database content.
Each server has node exporter configured to expose metrics on `9100` port, with `srv4` as the prometheus server.
`http://172.16.128.4:9090/targets` should show all targets as `up`.

### CODE REVIEW

**Comment by bdeferme** [comment]
As a suggestion for srv5 you could run grafana on it and add a simple dashboard to visualize some of the metrics.

File: src/poc/ansible/host_vars/srv1.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        aliases:
+          - grafana
+      - name: srv4
+        ip: 172.16.128.4
+        aliases:
+          - db
+      - name: srv5
+        ip: 172.16.128.5
+        aliases:
+          - base
```
Comment (bdeferme) [inline src/poc/ansible/host_vars/srv1.yml]: This doesn't match up with the table in your comment. I guess the table is out of date?

srv3 and srv4 are flipped around in the table (or here). Is srv3/4 running prometheus and grafana? In that case ignore my comment on srv5 potentially being a grafana node.

If that's the case maybe srv5 can be:

- A log collector (also pretty common)
  - Loki
  - Graylog
  - Logstash
  - Rsyslog
  - ...
- A haproxy node (reverse proxy, loadbalancing), also quite common
  - ELK stack

File: src/poc/ansible/main.yml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    - name: Restart service mariadb, in all cases
+      ansible.builtin.service:
+        name: mariadb
+        state: restarted
+
+- name: Configure srv4
+  hosts: srv4
+  roles:
+    - prometheus.prometheus.prometheus
+
```
Comment (bdeferme) [inline src/poc/ansible/main.yml:86]: Prometheus should probably be configured to scrape all target nodes.

File: src/poc/ansible/host_vars/srv1.yml (hunk 1 above)
Comment (AntonVanAssche) [reply src/poc/ansible/host_vars/srv1.yml]: Currently srv4 only runs Prometheus and Node Exporter. I could add Grafana to it, but I considering installing it on srv5 paired with a samba share. I just have to find up-to-date roles to help me with this.

See: 2059dff901ab57393858c2da116855b444d568d0

**Comment by AntonVanAssche** [comment]
> As a suggestion for srv5 you could run grafana on it and add a simple dashboard to visualize some of the metrics.

I was thinking of pairing Grafana on srv5 with a Samba share. I believe both are commonly used in the industry.

Grafana could also be installed on srv4 (instead of srv5), which might be a bit more logical.

I haven't decided yet what to do. I'll definitely look at your other suggestions first.

>   - A log collector (also pretty common)
>       - Loki
>       - Graylog
>       - Logstash
>       - Rsyslog
>       - ...
>   - A haproxy node (reverse proxy, loadbalancing), also quite common
>       - ELK stack

File: src/poc/ansible/main.yml (hunk 2 above)
Comment (AntonVanAssche) [reply src/poc/ansible/main.yml:86]: See: 222bf23

File: src/poc/ansible/host_vars/srv1.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        aliases:
+          - db
+      - name: srv4
+        ip: 172.16.128.4
+        aliases:
+          - prometheus
+      - name: srv5
+        ip: 172.16.128.5
+        aliases:
+          - grafana
```
Comment (AntonVanAssche) [inline src/poc/ansible/host_vars/srv1.yml:43]: Should be `samba` instead of `grafana`.

**Pull request merged**